### PR TITLE
[libzip] update to rel-1-6-1

### PR DESCRIPTION
**Pull request by mitzal**
Libzip updated to rel-1-6-1
[ #10778 ]

### CODE REVIEW

File: ports/libzip/portfile.cmake [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
     REPO nih-at/libzip
-    REF rel-1-5-2
-    SHA512 5ba765c5d4ab47dff24bfa5e73b798046126fcc88b29d5d9ce9d77d035499ae91d90cc526f1f73bbefa07b7b68ff6cf77e912e5793859f801caaf2061cb20aee
+    REF rel-1-6-1
+    SHA512 7ee414c063f9f76bec7d96ff9dadbc4be8d37a7b907b977882bf40f8ab66f0e46d3b8f70083c7bd272cc298d855d0d72b494b5772f26e1f4ff7ffeefe780adaf
     HEAD_REF master
-	PATCHES avoid_computation_on_void_pointer.patch
 )
 
 # AES encryption
```
Comment (LilyWangL) [inline ports/libzip/portfile.cmake:9]:
```
set(USE_OPENSSL OFF)
if("openssl" IN_LIST FEATURES)
    set(USE_OPENSSL ON)
endif()
set(USE_BZIP2 OFF)
if("bzip2" IN_LIST FEATURES)
    set(USE_BZIP2 ON)
endif()
```
The way to handle feature can be replaced by [vcpkg_check_features()](https://github.com/microsoft/vcpkg/blob/a3a6530631df905eb5c0e26d0b20d7d548e0c465/docs/maintainers/vcpkg_check_features.md).

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/libzip/portfile.cmake (hunk 1 above)
Comment (mitzal) [reply ports/libzip/portfile.cmake]: changed

**Review by LilyWangL — APPROVED** [review]

File: ports/libzip/portfile.cmake [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    openssl OPENSSL
+    bzip2 BZIP2
+)
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS
+        ${FEATURE_OPTIONS}
```
Comment (janisozaur) [inline ports/libzip/portfile.cmake:19]: The options are [`ENABLE_OPENSSL` and `ENABLE_BZIP2` in libzip's code](https://github.com/nih-at/libzip/blob/5487dcae97291c5eb598ea129aff045e1e2eeb65/CMakeLists.txt#L13-L16), but I see no mapping here between `OPENSSL` and `ENABLE_OPENSSL`. @mitzal is that intended and correct?

**Review by janisozaur — CHANGES_REQUESTED** [review]
Please merge https://github.com/mitzal/vcpkg/pull/1

File: ports/libzip/portfile.cmake (hunk 2 above)
Comment (mitzal) [reply ports/libzip/portfile.cmake:20]: Should be ENABLE_OPENSSL, you're right; thanks for changes.

**Comment by ras0219-msft** [comment]
Looking at the upstream code[1], it seems like only one of the crypto backends will be used. Therefore, it seems to me that we should restructure the features as:
```
Default-Features: default-aes
...
Feature: default-aes
Build-Depends: libzip[core,openssl] (!windows), libzip[core,wincrypto] (windows)

Feature: openssl
Build-Depends: openssl

Feature: wincrypto
```

We also need to explicitly disable the other backends (mbedTLS, gnutls, and commoncrypto) to ensure they aren't accidentally linked. These could be optionally made into features in the future if desired.

[1] https://github.com/nih-at/libzip/blob/8bd4f66d906026ab85bd2d05e47c6be0e144135f/CMakeLists.txt#L259

**Comment by mitzal** [comment]
> Looking at the upstream code[1], it seems like only one of the crypto backends will be used. Therefore, it seems to me that we should restructure the features as:
> 
> ```
> Default-Features: default-aes
> ...
> Feature: default-aes
> Build-Depends: libzip[core,openssl] (!windows), libzip[core,wincrypto] (windows)
> 
> Feature: openssl
> Build-Depends: openssl
> 
> Feature: wincrypto
> ```
> 
> We also need to explicitly disable the other backends (mbedTLS, gnutls, and commoncrypto) to ensure they aren't accidentally linked. These could be optionally made into features in the future if desired.
> 
> [1] https://github.com/nih-at/libzip/blob/8bd4f66d906026ab85bd2d05e47c6be0e144135f/CMakeLists.txt#L259

Features have been restructured and some additional options have been added.

**Comment by janisozaur** [comment]
https://dev.azure.com/vcpkg/public/_build/results?buildId=34181&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=a4822bc1-f3a9-5984-c941-6c1b7aa2ab81&l=18

`|libzip   |Fail|core|Test passes in baseline but fails in current run. If expected update ci.baseline.txt with 'libzip:x64-linux=fail'`

**Comment by ras0219-msft** [comment]
LGTM, thanks for the fixes!

**Pull request merged**